packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx

```diff
@@ -136,25 +138,57 @@ export const ChatListItem = ({ chat, myId, isSelected, isRead }: IMgtChatListIte
   // if chat changes, update the internal state to match
   useEffect(() => {
     setChatInternal(chat);
+    if (isLoaded()) {
+      checkWhetherToMarkAsRead(chat);
+    }
   }, [chat]);
 
   // enrich the chat if necessary
   useEffect(() => {
-    if (chatInternal.id && (!chatInternal.chatType || !chatInternal.members)) {
+    if (isLoaded()) {
       const provider = Providers.globalProvider;
       if (provider && provider.state === ProviderState.SignedIn) {
         const graph = provider.graph.forComponent('ChatListItem');
         const load = (id: string): Promise<Chat> => {
           return loadChatWithPreview(graph, id);
         };
-        load(chatInternal.id).then(
-          c => setChatInternal(c),
+        load(chatInternal.id!).then(
+          c => {
+            setChatInternal(c);
+            checkWhetherToMarkAsRead(c);
+          },
           e => error(e)
         );
       }
     }
   }, [chatInternal]);
 
+  const isLoaded = () => {
+    return chatInternal.id && (!chatInternal.chatType || !chatInternal.members);
+  };
+
+  // check whether to mark the chat as read or not
+  const checkWhetherToMarkAsRead = (c: Chat) => {
+    cache.loadLastReadTime(c.id!).then(lastReadData => {
+      if (lastReadData) {
+        const lastUpdatedDateTime = new Date(c.lastUpdatedDateTime as string);
+        const lastMessagePreviewCreatedDateTime = new Date(c.lastMessagePreview?.createdDateTime as string);
+        const lastReadTime = new Date(lastReadData.lastReadTime as string);
+        if (
+          lastUpdatedDateTime > lastReadTime ||
+          lastMessagePreviewCreatedDateTime > lastReadTime ||
+          !lastReadData.lastReadTime
+        ) {
+          log('marking chat as unread: ', c.id);
+          setRead(false);
```

Comment:
just a suggestion the "if" can be pass into setRead so you don't explictly need to say true or false

Reply:
Yeah that works, I think we lose the logging that way. Which I guess is fine since I'm done testing it